### DISCUSSION TITLE
location schema

### DISCUSSION THREAD

**Discussion by Easton97-Jens** [ZoeyVid/NPMplus#618] [Q&A] [answered]
Kann man irgendwo das Schema anpassen, damit man noch andere Möglichkeiten außer HTTP und HTTPS verwenden kann?

**Comment by Zoey2936** [comment]
Wie meinst du?

**Reply by Easton97-Jens** [reply]
dass man ssh: oder auch mysql: auf den dahinterliegenden Rechner zugreifen kann

**Reply by Easton97-Jens** [reply]
Um die Verbindung mit der echten IP durchreichen zu können, um die Sicherheit dann auf dem Rechner anpassen zu können, um die Zugriffe besser zu protokollieren und Abwehrmaßnahmen zu treffen.

**Reply by Easton97-Jens** [reply]
Ihr ein Beispiel.
    location /ssh/ {
        proxy_pass ssh://your_ssh_server:22;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }

**Reply by Zoey2936** [reply]
Also tut mir leid, aber ist das ganze ein Witz? Falls nein, http(s) und ssh sind nicht untereinander kompatibel

**Reply by Zoey2936** [reply]
Ich kann dir fail2ban empfehlen oder du nutzt crowdsec mit ssh collection und iptables bouncer

**Reply by Easton97-Jens** [reply]
Das ist mir bekannt. Ich möchte nicht jeden PC im Router freigeben und nicht jedes Mal die Konfiguration ändern. Ich kann das mit einem Stream nicht so realisieren, wie ich es möchte. das ganze läuft über proxy_protocol modul

**Reply by Zoey2936** [reply]
heißt du brauchst proxy protokoll in npmplus?

**Reply by Easton97-Jens** [reply]
Ja, das bräuchte ich dazu. Es ist auch schon ein Teil aktiv, und zwar with-stream_realip_module, was den Stream angeht. Da ich das in einem Stream aktivieren konnte, um die echte IP über das proxy_protocol in der Stream-Konfiguration an den PC dahinter weiterleiten zu können. Es geht halt jetzt um spezielle Fälle, die das proxy_protocol nicht verarbeiten kann. Die Grundlage dafür bildet nginx-quic. Die meisten erforderlichen Voraussetzungen sollten bereits erfüllt sein, damit dies reibungslos funktionieren kann.

**Reply by Easton97-Jens** [reply]
frage brauchst du hilfe oder unterstützung bei der umsetzung?

**Reply by Zoey2936** [reply]
was willst du jetzt nochmal genau tun?

**Reply by Easton97-Jens** [reply]
Das ist in der location das schema ändern kann auf das was ich für meine Projekte brauche da es immomentnur http und https gibt.
Ist das verständlich? Sonst könnte ich dir das über discord reden und dir das zeigen wenn es für dich einfach ist

**Reply by Zoey2936** [reply]
das geht nicht, wie oft den noch

**Reply by Zoey2936** [reply, marked as answer]
du kannst maximal einen stream erstellen und dann dich mittels proxy protokoll zu einem ssh server der per proxy protkoll erreichbar ist dich verbinden und das ganze über tcp exposen
